src/live_restore/live_restore_server.c

```diff
@@ -105,7 +105,13 @@ __live_restore_worker_stop(WT_SESSION_IMPL *session, WT_THREAD *ctx)
     server->threads_working--;
 
     if (server->threads_working == 0) {
-        if (!F_ISSET(S2C(session), WT_CONN_CLOSING)) {
+        /*
+         * If the migration running flag is set we're in this function due to the natural completion
+         * of background migration. The queue must be empty and we should clean up live restore
+         * files. If the flag isn't set it's because we're closing the connection. We can't assume
+         * the migration queue is empty so skip these checks.
+         */
+        if (F_ISSET(S2C(session), WT_CONN_READY)) {
```

Comment:
If we check this on startup and sleep until it is set isn't it guaranteed here?

Reply:
I missed updating the comment in the last commit, but the behaviour is the same as with the `MIGRATION_RUNNING` flag. `WT_CONN_READY` is cleared at the very beginning of `__conn_close` and is actually a better "connection closing" flag than `WT_CONN_CLOSING` is